src/gsy_e/models/strategy/smart_meter.py

```diff
@@ -249,8 +249,10 @@ def event_bid_traded(self, *, market_id, bid_trade):
         super().event_bid_traded(market_id=market_id, bid_trade=bid_trade)
 
         market = self.area.get_spot_or_future_market_by_id(market_id)
-        self.state.decrement_energy_requirement(
-            bid_trade.traded_energy * 1000, market.time_slot, self.owner.name)
+        self._energy_params.decrement_energy_requirement(
+            energy_kWh=bid_trade.traded_energy,
```

Comment:
The previous code was decrementing energy using kWh units (`bid_trade.traded_energy * 1000`). However, the `decrement_energy_requirement()` method in the State class should take Wh instead. In this PR, this is dealt with in the `SmartMeterEnergyParameters.decrement_energy_requirement()` method (where kWh are multiplied by 1000).